src/std/misc/func.ss

```diff
@@ -202,3 +209,32 @@
   (if limit
     (if (> limit 0) test (lambda (_) #f))
     pred))
+
+;; pred-after-sequence returns a predicate which returns #t after a
+;; matching sequence. True is returned limit times, if limit is not #f.
+;;
+;; Example:
+;;  (filter (pred-after-sequence [1 2]) [1 2 'a 1 2 'b]) => (a b)
+(def (pred-after-sequence list (limit #f))
+  (declare (not safe) (fixnum))
+  ;; TODO when contracts merge
+  ;; (@contract (pred-sequence list limit) (or (not limit) (fixnum? limit)))
+  (if (pair? list)
+    (let ((i 0) (len (length list)) (is-after #f))
+      (unless (fixnum? limit) (set! limit -1))
+      (lambda (v)
+        (match* (is-after limit)
+          ((#t _) (set! is-after #f) #t) ; success
+          ((_ 0)  #f)                    ; limit reached
+          (else                          ; check
+            (and
+              (or (equal? (list-ref list i) v)
+                  (begin (set! i 0) #f))
+              (set! i (1+ i))
+              (= len i)
+              ;; the sequence matched
+              (set! i 0)
+              (set! is-after #t)
+              (when (> limit 0) (set! limit (1- limit)))
```

Comment:
it's very confusing to have mutations inside an `and`. Can we rewrite this to be cleaner?